src/agentlab/llm/huggingface_utils.py

```diff
@@ -40,14 +40,17 @@ class HFBaseChatModel(AbstractChatModel):
         description="The number of times to retry the server if it fails to respond",
     )
 
-    def __init__(self, model_name, n_retry_server):
+    def __init__(self, model_name, base_model_name, n_retry_server):
         super().__init__()
         self.n_retry_server = n_retry_server
 
-        self.tokenizer = AutoTokenizer.from_pretrained(model_name)
+        if base_model_name is None:
+            self.tokenizer = AutoTokenizer.from_pretrained(model_name)
+        else:
+            self.tokenizer = AutoTokenizer.from_pretrained(base_model_name)
```

Comment:
I don't get what's the point of that...

Reply:
yes, sorry a little cryptic. I made this to support LoRA checkpoints, which are stored in the `model_name` directory. They are applied to a base model, stored at `base_model_name` directory.